### PR TITLE
Add movie overview and detail pages

### PR DESCRIPTION
**Pull request by nielsrowinbik**

### CODE REVIEW

File: src/components/MainPosterList.vue [hunk 1]
```diff
@@ -1,14 +1,38 @@
...
 
 <style lang="postcss" scoped>
-.grid > *,
-.grid img {
-    @apply rounded-lg;
+.wrapper {
+    background-position: center -50vh;
+    background-repeat: no-repeat;
+    /* TODO: Use dynamic image */
+    background-image: url(https://www.themoviedb.org/t/p/w600_and_h900_bestv2/9gk7adHYeDvHkCSEqAvQNLV5Uge.jpg);
```
Comment (iamrgroot) [inline src/components/MainPosterList.vue]: I think you need to add a CSS variable like [here](https://www.telerik.com/blogs/passing-variables-to-css-on-a-vue-component)

File: src/components/NavButton.vue [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<template>
+    <router-link
+        :class="['px-2', 'py-2', 'rounded-full', 'text-white', 'items-center', 'justify-center', { 'bg-black': variant === 'default' }, { 'bg-opacity-40': variant === 'default' }, {'hover:bg-black': variant === 'transparent'}, {'hover:bg-opacity-40': variant === 'transparent'}, {'hover:text-white': variant === 'transparent'}]"
```
Comment (iamrgroot) [inline src/components/NavButton.vue:3]: Maybe at a certain point add some line breaks?

File: src/main.ts [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
+import routes from 'voie-pages';
 import { createRouter, createWebHistory } from 'vue-router';
 
 import { useMirageJS } from './plugins/miragejs';
 
 import './index.css';
 import App from './App.vue';
 
-const router = createRouter({
+export const router = createRouter({
```
Comment (iamrgroot) [inline src/main.ts]: Router has own plugin file now

File: src/pages/detail/movie/[id].vue [hunk 4]
```diff
@@ -0,0 +1,71 @@
+<template>
+    <Nav>
+        <NavButton to="/overview/movies">
+            <!-- TODO: Extract icon to its own component  -->
```
Comment (iamrgroot) [inline src/pages/detail/movie/[id].vue]: Maybe use @mdi/svg npm package?

File: src/components/MainPosterList.vue (hunk 1 above)
Comment (nielsrowinbik) [reply src/components/MainPosterList.vue]: Yep, that's how I've done it now!

File: src/main.ts (hunk 3 above)
Comment (nielsrowinbik) [reply src/main.ts]: Should get fixed when main is merged back into this branch.

File: src/components/NavButton.vue (hunk 2 above)
Comment (nielsrowinbik) [reply src/components/NavButton.vue:3]: I was expecting ESLint to do that for me. I hate this particular line, any suggestions on making this more readable (other than adding line breaks)? Or is line breaks the way to go?

Comment (iamrgroot) [reply src/components/NavButton.vue:3]: Maybe adding [this rule](https://eslint.org/docs/rules/max-len) helps

File: src/pages/detail/movie/[id].vue (hunk 4 above)
Comment (nielsrowinbik) [reply src/pages/detail/movie/[id].vue]: Ended up using `@mdi/js` and adding a custom Icon component, but this has been addressed.

**Review by iamrgroot — APPROVED** [review]

**Pull request merged**